.github/workflows/gh-ci-cron.yaml

```diff
@@ -264,14 +264,16 @@ jobs:
   pypi-latest-release:
     # A set of runner to check that the latest conda release works as expected
     if: "github.repository == 'MDAnalysis/mdanalysis'"
-    runs-on: ${{ matrix.os }}-latest
+    runs-on: ${{ matrix.os }}
     timeout-minutes: 60
     strategy:
         fail-fast: false
         matrix:
-          os: [ubuntu, macos, windows]
+          os: [ubuntu-latest, macos-latest, macos-14, windows-latest]
           python-version: ["3.9", "3.10", "3.11", "3.12"]
           wheels: ['true', 'false']
+          exclude:
+          - {os: "macos-14", python-version: "3.9"}
```

Comment:
why is this being disabled? as show in #4445 we should be able to build on 3.9 with macos-14

Reply:
>   Version 3.9 was not found in the local cache
>  Error: The version '3.9' with architecture 'arm64' was not found for macOS 14.2.1.
>  The list of all available versions can be found here: https://raw.githubusercontent.com/actions/python-versions/main/versions-manifest.json

I got this failure. Looking at the `versions-manifest.json` as suggested, it looks like all `3.9` is not provided with `arch: arm64`.

Reply:
That's.. odd. I've not glanced upstream, but we should probably keep an eye on updates - it's very possible they couldn't backport stuff properly until they had their own CI runners for this.